providers/nextcloud_test.go

```diff
@@ -96,8 +98,9 @@ func TestNextcloudProviderGetEmailAddress(t *testing.T) {
 	p.ValidateURL.Path = userPath
 	p.ValidateURL.RawQuery = formatJSON
 
+	ctx := context.TODO()
```

Comment:
Please do the others in this file too
```suggestion
	ctx := context.Background()
```